pandas/core/arrays/numeric.py

```diff
@@ -231,16 +232,17 @@ def _coerce_to_data_and_mask(
             values = np.ones(values.shape, dtype=dtype)
         else:
             idx = np.nanargmax(values)
-            if int(values[idx]) != original[idx]:
-                # We have ints that lost precision during the cast.
-                inferred_type = lib.infer_dtype(original, skipna=True)
-                if (
-                    inferred_type not in ["floating", "mixed-integer-float"]
-                    and not mask.any()
-                ):
-                    values = np.asarray(original, dtype=dtype)
-                else:
-                    values = np.asarray(original, dtype="object")
+            if not (pd.isna(values[idx]) or pd.isna(original[idx])):
```

Comment:
can you check this without pd.isna?  we try to avoid importing pd in files that are "high up" in the dependency hierarchy